fvm/errors/base.go

```diff
@@ -0,0 +1,230 @@
+package errors
+
+import (
+	"errors"
+	stdErrors "errors"
+	"fmt"
+
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// InvalidAddressError indicates that a transaction references an invalid flow Address
+// in either the Authorizers or Payer field.
+type InvalidAddressError struct {
+	address flow.Address
+	err     error
+}
+
+// NewInvalidAddressErrorf constructs a new InvalidAddressError
+func NewInvalidAddressErrorf(msg string, err error, address flow.Address) *InvalidAddressError {
+	return &InvalidAddressError{err: fmt.Errorf(msg, err), address: address}
+}
```

Comment:
```suggestion
func NewInvalidAddressErrorf(address flow.Address, msg string, args ...interface{}) *InvalidAddressError {
	return &InvalidAddressError{err: fmt.Errorf(msg, args...), address: address}
```

As a pattern I would suggest to format these like `New....Errorf(errorSpecificArgs..., msg string, format interface{}...)`. That way the final `n` arguments mirror the inputs to `fmt.Errorf`. The `error` input can always be provided as part of the format arguments (or can be omitted if there is no child error), so it matches the wrapping pattern in regular untyped `fmt.Errorf`. `fmt.Errorf("some error: %w", err)` and `errors. NewInvalidAddressErrorf(address, "address invalid because xyz: %w", err)`. This way the caller can decide how to add context to the error and isn't constrained to passing a specific format string as this would require (one `%w` and no other format variables).

Reply:
I see, I changed all of the errors with formating options.